Assets/MixedRealityToolkit.Examples/Experimental/Audio/Scripts/MicrophoneAmplitudeDemo.cs

```diff
@@ -0,0 +1,208 @@
+        /// </summary>
+        private Material VisibleMaterial
+        {
+            get
+            {
+                if (visibleMaterial == null)
+                {
+                    visibleMaterial = SpatialMeshObserver?.VisibleMaterial;
+                }
+                return visibleMaterial;
+            }
+        }
+
+        /// <summary>
+        /// Class providing microphone stream management support on Microsoft Windows based devices.
+        /// </summary>
+        private WindowsMicrophoneStream micStream = null;
+
+        /// <summary>
+        /// The average amplitude of the sound captured during the most recent microphone update.
+        /// </summary>
+        private float averageAmplitude = 0.0f;
+
+        #if UNITY_EDITOR
+        /// <summary>
+        /// Cached material values used to restore initial settings when running the demo in the editor.
+        /// </summary>
+        private Color defaultMaterialColor = Color.black;
+        private int defaultWireThickness = 0;
+#endif // UNITY_EDITOR
+
+        private void Awake()
+        {
+            // We do not wish to play the ambient room sound from the audio source.
+            gameObject.GetComponent<AudioSource>().volume = 0.0f;
+
+            if (VisibleMaterial != null)
+            {
+#if UNITY_EDITOR
+                // Cache the initial material settings.
+                defaultMaterialColor = VisibleMaterial.GetColor("_WireColor");
+                defaultWireThickness = VisibleMaterial.GetInt("_WireThickness");
+#endif // UNITY_EDITOR
+
+                VisibleMaterial.SetColor("_WireColor", Color.blue);
+            }
+
+            micStream = new WindowsMicrophoneStream();
+            micStream.Gain = 1.0f;
+
+            // Initialize the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.Initialize(WindowsMicrophoneStreamType.HighQualityVoice);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to initialize the microphone stream. {result}");
+            }
+
+            // Start the microphone stream.
+            // Do not keep the data and do not preview.
+            result = micStream.StartStream(false, false);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to start the microphone stream. {result}");
+            }
+        }
+
+        private void OnDestroy()
+        {
+            // Stop the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.StopStream();
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to stop the microphone stream. {result}");
+            }
+
+            // Uninitialize the microphone stream.
+            micStream.Uninitialize();
+            micStream = null;
```

Comment:
I think this should only happen if we succeeded in Awake() - otherwise this will just nullreference

Reply:
sounds good

Reply:
fixed